packages/expo-updates/src/ExponentUpdates.ts

```diff
@@ -0,0 +1,108 @@
+import { RCTDeviceEventEmitter } from '@unimodules/core';
+import Constants from 'expo-constants';
+import { EventEmitter, EventSubscription } from 'fbemitter';
+import { NativeModules } from 'react-native';
+
+import {
+  UpdateCheckResult,
+  UpdateFetchResult,
+  Listener,
+  UpdateEvent,
+  UpdateEventType,
+} from './Updates.types';
+
+export default NativeModules.ExponentUpdates
+  ? {
+      get name(): string {
+        return 'ExponentUpdates';
+      },
+      get manifest(): any {
+        return Constants.manifest;
+      },
+      async reload(): Promise<void> {
+        await NativeModules.ExponentUpdates.reloadFromCache();
+      },
+      async checkForUpdateAsync(): Promise<UpdateCheckResult> {
+        const result = await NativeModules.ExponentUpdates.checkForUpdateAsync();
+        if (!result) {
+          return { isAvailable: false };
+        }
+
+        return {
+          isAvailable: true,
+          manifest: typeof result === 'string' ? JSON.parse(result) : result,
+        };
+      },
+      async fetchUpdateAsync(): Promise<UpdateFetchResult> {
+        _isFetchingUpdate = true;
+        const result = await NativeModules.ExponentUpdates.fetchUpdateAsync();
+        setTimeout(() => {
+          _isFetchingUpdate = false;
+        }, 1000);
```

Comment:
What's this timeout for?

Reply:
Just in case something weird happens and the event comes through after the promise is resolved. I thought it would be better to be safe (other events should only ever happen if when the app is launched, so collision is unlikely) than to risk emitting an extra event. Do you think it's unnecessary? I can remove or replace with a shorter timeout if so.

Looking at the native code, the "send event" calls (which go directly through `RCTDeviceEventEmitter`) always happen immediately before the `promise.resolve` code.

Reply:
If the intent is to wait for one run of the event loop, `process.nextTick` or `setTimeout(..., 0)` communicates that better.